backend/src/test/java/ch/puzzle/pcts/service/business/CertificateCalculationBusinessServiceTest.java

```diff
@@ -0,0 +1,196 @@
+package ch.puzzle.pcts.service.business;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import ch.puzzle.pcts.exception.PCTSException;
+import ch.puzzle.pcts.model.calculation.Calculation;
+import ch.puzzle.pcts.model.calculation.certificatecalculation.CertificateCalculation;
+import ch.puzzle.pcts.model.certificate.Certificate;
+import ch.puzzle.pcts.model.certificatetype.CertificateKind;
+import ch.puzzle.pcts.model.certificatetype.CertificateType;
+import ch.puzzle.pcts.service.persistence.CertificateCalculationPersistenceService;
+import ch.puzzle.pcts.service.validation.CertificateCalculationValidationService;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CertificateCalculationBusinessServiceTest {
+
+    private static final Long ID = 1L;
+
+    @Mock
+    private CertificateCalculationValidationService validationService;
+
+    @Mock
+    private CertificateCalculationPersistenceService persistenceService;
+
+    @InjectMocks
+    private CertificateCalculationBusinessService businessService;
+
+    @Test
+    @DisplayName("Should get certificate calculations by calculation id")
+    void shouldGetByCalculationId() {
+        CertificateCalculation cc = mock(CertificateCalculation.class);
+        when(persistenceService.getByCalculationId(ID)).thenReturn(List.of(cc));
+
+        List<CertificateCalculation> result = businessService.getByCalculationId(ID);
+
+        assertEquals(1, result.size());
+        verify(persistenceService).getByCalculationId(ID);
+    }
+
+    @Test
+    @DisplayName("Should get certificate calculations by certificate id")
+    void shouldGetByCertificateId() {
+        CertificateCalculation cc = mock(CertificateCalculation.class);
+        when(persistenceService.getByCertificateId(ID)).thenReturn(List.of(cc));
+
+        List<CertificateCalculation> result = businessService.getByCertificateId(ID);
+
+        assertEquals(1, result.size());
+        verify(persistenceService).getByCertificateId(ID);
+    }
+
+    @Test
+    @DisplayName("Should calculate certificate points correctly")
+    void shouldCalculateCertificatePoints() {
```

Comment:
please 
this is prob the single most important ticket in the entire application
use parametrized test with null, 0 and any other value you can think of to ensure the main functionality is not broken